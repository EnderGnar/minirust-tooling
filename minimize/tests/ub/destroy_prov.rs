include!("../helper/transmute.rs");

fn main() { unsafe {
    let b = 2;
    let x = &b as *const i32; // valid ptr!

    // transmute round-trip
    let x = transmute::<*const i32, usize>(x);
    let x = transmute::<usize, *const i32>(x);

    let _x = *x; // invalid ptr!
} }
