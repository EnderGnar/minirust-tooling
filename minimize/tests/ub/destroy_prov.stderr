UB: non-zero-sized access with invalid pointer
